src/components/checkbox/checkbox.jsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import FormControl from '../form-control/form-control.jsx';
+import useMounted from '../../hooks/use-mounted.js';
+import Icon from '../icon/icon.jsx';
+import styles from './checkbox.css';
+import cautionSvg from '../../svgs/caution.svg';
+
+const Checkbox = forwardRef(function Checkbox(props, forwardedRef) {
+  const fallbackRef = useRef();
+  const ref = forwardedRef || fallbackRef;
+  const inputRef = useRef();
+  const mounted = useMounted();
+  const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+
+  useEffect(() => {
```

Comment:
Since this hook relies on a _rendered_ DOM node, we should switch this to

```suggestion
  useLayoutEffect(() => {
```

Reply:
And remove the mounted check since we know we have a DOM node.

Reply:
4e2f968353519f3dcbc9eec5298d64aa8c0ee459